### PR TITLE
Correct stable and preview upgrade codes for `MSI`

### LINKED ISSUE #13035: PowerShell 7.1.0-preview4 x64 executable removes PowerShell 7.0.2 and does not install preview4
## Steps to reproduce
- Install PowerShell 7.1.0-preview3 (7.1.0.3) [`PowerShell-7.1.0-preview.3-win-x64.msi`]
- Install PowerShell 7.0.2 [`PowerShell-7.0.2-win-x64.msi`]
- Verify that both appear separately in `C:\Program Files\PowerShell\`
- Install PowerShell 7.1.0-preview4 (7.1.0.4) [`PowerShell-7.1.0-preview.4-win-x64.msi`]
- Verify `C:\Program Files\PowerShell\7\` and `C:\Program Files\PowerShell\7-preview\` again

## Expected behavior

PowerShell Preview is updated to 7.1.0-preview4 and PowerShell 7.0.2 is not deleted.

## Actual behavior

PowerShell 7.0.2 is being deleted from `C:\Program Files\PowerShell\7\`
PowerShell 7.1.0-preview4 is not installed (but its installation is added to registry)
PowerShell 7.1.0-preview3 is installed.
image

## Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.1.0-preview.3
PSEdition                      Core
GitCommitId                    7.1.0-preview.3
OS                             Microsoft Windows 10.0.20150
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

# Additional info
This happened on my main PC (Windows 10 2004, build 19041.329) as well as I was able to reproduce it in VM with Windows 10 Insider Preview (build 20150.1000)
Also latest preview installation executable does not appear to have digital signature attached in comparison to the previous preview.
Something for @TravisEz13 I guess?

### PR DESCRIPTION
**Pull request by TravisEz13**
Fixes #13035 

# PR Summary

Correct stable and preview upgrade codes for msi

## PR Context

stable and preview were reversed in:
https://github.com/PowerShell/PowerShell/pull/12792

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: assets/Product.wxs [hunk 1]
```diff
@@ -9,17 +9,17 @@
   <!-- UpgradeCode GUID MUST REMAIN SAME THROUGHOUT ALL VERSIONS, otherwise, updates won't occur. -->
   <?if $(sys.BUILDARCH)=x64?>
     <?define ExplorerContextMenuDialogText = "&$(var.ProductName) $(env.SimpleProductVersion)"?>
-    <?define UpgradeCodePreview = "31ab5147-9a97-4452-8443-d9709f0516e1"?>
-    <?define UpgradeCodeRelease = "39243d76-adaf-42b1-94fb-16ecf83237c8"?>
+    <?define UpgradeCodePreview = "39243d76-adaf-42b1-94fb-16ecf83237c8"?>
```
Comment (TravisEz13) [inline assets/Product.wxs:12]: preview upgrade codes
```
x64 - 39243d76-adaf-42b1-94fb-16ecf83237c8	
x86 - 86abcfbd-1ccc-4a88-b8b2-0facfde29094
```
release upgrade codes
```
"UpgradeCodeX64", '31ab5147-9a97-4452-8443-d9709f0516e1', "Process")	
"UpgradeCodeX86", '1d00683b-0f84-4db8-a64f-2f98ad42fe06', "Process")
```

Comment (daxian-dbw) [reply assets/Product.wxs:12]: I'm curious where does `var.IsPreview` declared?

Comment (TravisEz13) [reply assets/Product.wxs:12]: https://github.com/PowerShell/PowerShell/pull/12792/files#diff-edfc56f59d77ed8604989ab0efbff45cR3060

**Review by daxian-dbw — APPROVED** [review]

**Review by heaths — APPROVED** [review]
Sorry about that. I tested all the upgrade scenarios in a sandbox using generated MSIs, but should've also tested against then-released MSIs.

File: test/packaging/windows/msi.tests.ps1 [hunk 2]
```diff
@@ -80,6 +80,33 @@ Describe -Name "Windows MSI" -Fixture {
         }
     }
 
+    Context "Upgrade code" {
+        BeforeAll {
+            $previewUpgladeCode = '39243d76-adaf-42b1-94fb-16ecf83237c8'
```
Comment (iSazonov) [inline test/packaging/windows/msi.tests.ps1:85]: If we add tests for Preview why not for Release?

If we add tests for x64 should we skip the tests on x86?

**Comment by minoseah629** [comment]
in a similar note, trying to install 7.0.2, it thinks there is a higher version already: 
image

**Comment by adityapatwardhan** [comment]
@iSazonov ok to merge this?

File: test/packaging/windows/msi.tests.ps1 [hunk 3]
```diff
@@ -80,6 +80,33 @@ Describe -Name "Windows MSI" -Fixture {
...
+        }
```
Comment (rjmholt) [inline test/packaging/windows/msi.tests.ps1:86]:
```suggestion
        }

```

**Review by rjmholt — APPROVED** [review]

**Comment by TravisEz13** [comment]
> in a similar note, trying to install 7.0.2, it thinks there is a higher version already:

This is after installing preview.4?

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 30 minutes

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 30 minutes ago

**Comment by minoseah629** [comment]
just tried to reinstall 7.0.2 and still getting same message as before: https://github.com/PowerShell/PowerShell/pull/13036#issuecomment-650274956

**Comment by heaths** [comment]
If you installed the newer preview, you'll need to first uninstall it.

**Comment by minoseah629** [comment]
@heaths thanks.  that worked

**Pull request merged**